src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/DetailsRightGrid.tsx

```diff
@@ -24,8 +25,9 @@ import './details-right-grid.scss';
 const DetailsRightGrid: FC = () => {
   const { t } = useKubevirtTranslation();
   const { createModal } = useModal();
+  const cluster = useClusterParam();
   const [{ clusterDefaultStorageClass, sortedStorageClasses, virtDefaultStorageClass }, loaded] =
-    useDefaultStorageClass();
+    useDefaultStorageClass(cluster);
```

Comment:
Can you update this hook to get the cluster itself as well?

Reply:
I looked at the places where `useDefaultStorageClass` is used. It looks like we pass the VM cluster in one place. Maybe the approach we should take with hooks is to pull the cluster inside each hook that needs it, but take an optional `cluster` parameter that overrides that value if a cluster is passed in. That way, we minimize the places where we're passing the cluster value around while making hooks friendly for lists to use. This way will further decouple the cluster logic from the rest of the code base.

Reply:
done it